File: main_code/poc_kmp_sdks/ios/Sources/ViewModels/DetailsViewModel.swift
import Foundation
import SharedSDK

/// ViewModel for the Details view
/// Manages UI state and business logic for displaying individual item details
@MainActor
class DetailsViewModel: ObservableObject {
    @Published var user: User?
    @Published var product: Product?
    @Published var isLoading = false
    @Published var error: String?
    
    private let sharedSDK = SharedSDK()
    
    /// Loads an item (user or product) by ID and type
    func loadItem(itemId: String, itemType: DetailsView.ItemType) {
        Task {
            isLoading = true
            error = nil
            user = nil
            product = nil
            
            do {
                switch itemType {
                case .user:
                    let fetchedUser = try await sharedSDK.getUserById(userId: itemId)
                    if let fetchedUser = fetchedUser {
                        user = fetchedUser
                    } else {
                        error = "User not found"
                    }
                    
                case .product:
                    let fetchedProduct = try await sharedSDK.getProductById(productId: itemId)
                    if let fetchedProduct = fetchedProduct {
                        product = fetchedProduct
                    } else {
                        error = "Product not found"
                    }
                }
                
                isLoading = false
            } catch {
                self.error = error.localizedDescription
                isLoading = false
            }
        }
    }
    
    /// Navigates back using the shared SDK
    func navigateBack() {
        sharedSDK.navigateBack()
    }
}

File: main_code/poc_kmp_sdks/android/src/androidMain/kotlin/ui/viewmodel/DetailsViewModel.kt
package ui.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import main_code.SharedSDK
import main_code.data.Product
import main_code.data.User

/**
 * ViewModel for the Details screen
 * Manages UI state and business logic for displaying individual item details
 */
class DetailsViewModel(
    private val sharedSDK: SharedSDK = SharedSDK()
) : ViewModel() {
    
    private val _uiState = MutableStateFlow(DetailsUiState())
    val uiState: StateFlow<DetailsUiState> = _uiState.asStateFlow()
    
    /**
     * Loads an item (user or product) by ID and type
     */
    fun loadItem(itemId: String, itemType: String) {
        viewModelScope.launch {
            _uiState.value = _uiState.value.copy(isLoading = true, error = null)
            
            try {
                when (itemType.lowercase()) {
                    "user" -> {
                        val user = sharedSDK.getUserById(itemId)
                        if (user != null) {
                            _uiState.value = _uiState.value.copy(
                                isLoading = false,
                                user = user,
                                product = null
                            )
                        } else {
                            _uiState.value = _uiState.value.copy(
                                isLoading = false,
                                error = "User not found"
                            )
                        }
                    }
                    
                    "product" -> {
                        val product = sharedSDK.getProductById(itemId)
                        if (product != null) {
                            _uiState.value = _uiState.value.copy(
                                isLoading = false,
                                product = product,
                                user = null
                            )
                        } else {
                            _uiState.value = _uiState.value.copy(
                                isLoading = false,
                                error = "Product not found"
                            )
                        }
                    }
                    
                    else -> {
                        _uiState.value = _uiState.value.copy(
                            isLoading = false,
                            error = "Invalid item type: $itemType"
                        )
                    }
                }
            } catch (e: Exception) {
                _uiState.value = _uiState.value.copy(
                    isLoading = false,
                    error = e.message ?: "Unknown error occurred"
                )
            }
        }
    }
    
    /**
     * Navigates back using the shared SDK
     */
    fun navigateBack() {
        sharedSDK.navigateBack()
    }
}

/**
 * UI state for the Details screen
 */
data class DetailsUiState(
    val isLoading: Boolean = false,
    val user: User? = null,
    val product: Product? = null,
    val error: String? = null
)